### PR TITLE
fix(docs): fix deprecation warning for boundActionCreators

### PR DESCRIPTION
**Pull request by muescha**
### Steps to reproduce: 
- when using `gatsby develop` on gatsby www repo

### Warning

- then this deprecation warning shows up:

```bash
boundActionCreators is deprecated. Please use actions instead. 
For migration instructions, see https://gatsby.dev/boundActionCreators

Check the following files:
⠀
src/pages/docs/actions.js
```

### Solution

it reveals that redux uses [`bindActionCreators`](https://redux.js.org/api/bindactioncreators) instead of `boundActionCreators` 

### Fix

- changed text to bindactioncreators
- added link to redux bindactioncreators 

### Fixes

- no more warning printed in terminal

### Documentation

n/a

## Related Issues

n/a

### CODE REVIEW

File: www/src/pages/docs/actions.js [hunk 1]
```diff
@@ -41,8 +41,13 @@ class ActionCreatorsDocs extends React.Component {
             <a href="http://redux.js.org">Redux</a>
             {` `}
             internally to manage state. When you implement a Gatsby API, you are
-            passed a collection of actions (equivalent to boundActionCreators in
-            redux) which you can use to manipulate state on your site.
+            passed a collection of actions (equivalent to
```
Comment (freiksenet) [inline www/src/pages/docs/actions.js]:
```suggestion
            passed a collection of actions (equivalent to actions bound with
```

**Comment by vladar** [comment]
Looks like the warning itself is printed in `develop` command where we scan all js files (including pages and components) and just check if file content includes API name string:

https://github.com/gatsbyjs/gatsby/blob/1b44f333de0a8c8768f925f99f4834e07bcb5fa7/packages/gatsby/src/commands/develop.js#L525-L533

While it makes sense for `pathContext` API, it produces false positives like this for `boundActionCreators` (which is a node only API).

I guess the workaround in this PR works but if we see other issues like this we should consider addressing the original source of this issue.

Leaving this comment just as a hint for future if we need it. It shouldn't affect or block this PR.

**Comment by muescha** [comment]
yes it just a plain text search - it search also in the comments or texts in a javascript file. 

@vladar should i add a comment with a link to your note here to the `develop.js#L525`?

**Comment by muescha** [comment]
the failing test comes from npm audit and is fixed now with 20157. 
how i can rerun the CI?

**Pull request merged**